### ISSUE TITLE
Week 5 ready for review

### ISSUE THREAD

**Issue by jpdhmc** [jpdhmc/encounter-tracker#2]
@pawaitemadisoncollege Week 5 is complete. I'm sorry for the late turn-in! I chose to implement a one-to-many relationship in my indie project.

What were your key learning points/takeaways?
I felt like Hibernate really "clicked" for me this week, and I got more comfortable in the intellij database view. Learning how to handle one-to-many relationships enabled a lot of work to be done on my indie project!

What challenged you?
I still find it hard to parse exactly how everything interacts when reading DAO classes. I relied very heavily on the demo code to write my DAOs.

What problems did you solve and what resources did you use to solve them?
Most things worked well for me this week. I was expecting mysql to give me more trouble but it was very smooth!

**Comment by pawaitemadisoncollege** [comment]
@jpdhmc Great job implementing this in your indie project - that sets you up really well to hit the major milestone of Checkpoint 2 this week!  You have what you need to start interacting with all of this data in the jsps, too!

- [ ] Think about testing the delete scenarios in one-to-many relationships more fully. For example, if a user is deleted, what should happen to their encounters? What if an encounter is deleted? Write tests to make sure whatever should happen, does happen.
- [ ] You'll likely notice, if you haven't already, that writing a Dao for each entity will be very repetitive - check out the  abstract/generic dao and plan to use it in your project so that you can use a single dao! Note that you will still have unit tests each for each entity(dao) to make sure that annotations and database relationships are set up correctly.
- [ ] Verifying the entities are saved/updated/inserted correctly can be more thorough by implementing and using equals() on each entity - refer to the video Comparing Hibernate Entities with equals in week 5 for more detail.  An example of using that equals method in a unit test would be something like `assertTrue(expectedEncounter.equals(actualEncounter));` so that you don't have to compare each instance var/column manually like this: https://github.com/jpdhmc/encounter-tracker/blob/8872303efa32a68f1a6146980e7704f708ce3e2b/src/test/java/jdh/persistence/EncounterDaoTest.java#L70-L71
